### PR TITLE
cleanup(common): avoid bigtable in unit tests

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #12485 

---

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12488)

### CODE REVIEW

File: google/cloud/completion_queue_test.cc [hunk 1]
```diff
@@ -32,40 +32,36 @@ namespace cloud {
...
+      (grpc::ClientContext*, Request const&, grpc::CompletionQueue* cq));
 
-  MOCK_METHOD(
-      std::unique_ptr<
-          ::grpc::ClientAsyncReaderInterface<btproto::ReadRowsResponse>>,
-      AsyncReadRows,
-      (grpc::ClientContext*, btproto::ReadRowsRequest const&,
-       grpc::CompletionQueue* cq));
+  MOCK_METHOD(std::unique_ptr<::grpc::ClientAsyncReaderInterface<Response>>,
+              AsyncReadRows,
```
Comment (dbolduc) [inline google/cloud/completion_queue_test.cc:55]: optional: s/ReadRows/ResponseStream/

File: google/cloud/internal/async_retry_unary_rpc_test.cc [hunk 2]
```diff
@@ -48,17 +50,15 @@ using ::testing::Return;
 class MockStub {
  public:
   MOCK_METHOD(
-      std::unique_ptr<grpc::ClientAsyncResponseReaderInterface<btadmin::Table>>,
+      std::unique_ptr<grpc::ClientAsyncResponseReaderInterface<Response>>,
       AsyncGetTable,
```
Comment (dbolduc) [inline google/cloud/internal/async_retry_unary_rpc_test.cc:54]: optional: s/Table/Response/

**Review by dbolduc — APPROVED** [review]

**Comment by dbolduc** [comment]
Thanks!

File: google/cloud/completion_queue_test.cc [hunk 3]
```diff
@@ -32,40 +32,36 @@ namespace cloud {
...
+using Response = google::protobuf::Timestamp;
+using MockResponseReader =
+    ::google::cloud::testing_util::MockAsyncResponseReader<Response>;
 
 class MockClient {
  public:
   MOCK_METHOD(
-      std::unique_ptr<grpc::ClientAsyncResponseReaderInterface<btadmin::Table>>,
+      std::unique_ptr<grpc::ClientAsyncResponseReaderInterface<Response>>,
       AsyncGetTable,
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/completion_queue_test.cc]: Done

**Pull request merged**